sycl-jit/jit-compiler/include/KernelFusion.h

```diff
@@ -61,13 +61,13 @@ class RTCResult {
   explicit RTCResult(const char *ErrorMessage)
       : Failed{true}, BundleInfo{}, ErrorMessage{ErrorMessage} {}
 
-  explicit RTCResult(RTCBundleInfo &&BundleInfo)
-      : Failed{false}, BundleInfo{std::move(BundleInfo)}, ErrorMessage{} {}
+  explicit RTCResult(RTCBundleInfo &&BundleInfo, const char *BuildLog)
+      : Failed{false}, BundleInfo{std::move(BundleInfo)}, ErrorMessage{
+                                                              BuildLog} {}
 
   bool failed() const { return Failed; }
 
   const char *getErrorMessage() const {
-    assert(failed() && "No error message present");
```

Comment:
Why are we dropping this?

Reply:
We use the error message field for the build log when compilation was successful, so we need to be able to access this field even if the result isn't failed.